models/cohortdata.go

```diff
@@ -130,20 +130,45 @@ func (h CohortData) RetrieveCohortOverlapStats(sourceId int, caseCohortId int, c
 func (h CohortData) RetrieveCohortOverlapStatsWithoutFilteringOnConceptValue(sourceId int, caseCohortId int, controlCohortId int,
 	otherFilterConceptIds []int64, filterCohortPairs []utils.CustomDichotomousVariableDef) (CohortOverlapStats, error) {
 
+	// special case for when filter lists are empty:
+	if len(otherFilterConceptIds) == 0 && len(filterCohortPairs) == 0 {
```

Comment:
I think really this is only relevant when `len(otherFilterConceptIds) == 0` if we only have dichotomous we still don't need to care about the observation table.

Reply:
this will be a bigger change, as we will have to adjust `QueryFilterByConceptIdsAndCohortPairsHelper` as well. In this method, we'll have to join on either `case_cohort` or `control_cohort`, leading to perhaps less readable code. Also the code will become harder to test, as it will branch off into yet another case. IMO we should consider tackling this if expect significant performance gains. Maybe we can run some tests to check this?